pkg/process/config/yaml_config.go

```diff
@@ -180,33 +178,6 @@ func (a *AgentConfig) LoadProcessYamlConfig(path string, canAccessContainers boo
 		config.Datadog.Set(key(ns, "cmd_port"), 6162)
 	}
 
-	// use `internal_profiling.enabled` field in `process_config` section to enable/disable profiling for process-agent,
-	// but use the configuration from main agent to fill the settings
-	if config.Datadog.IsSet(key(ns, "internal_profiling.enabled")) {
```

Comment:
We should use `GetBool` instead of `IsSet`. If the user puts
```
process_config:
  internal_profiling:
    enabled: false
```
in the `datadog.yaml` file, `IsSet`will return `true` but we don't want profiling enabled.

Reply:
Nice catch! That is such weird behavior 😆